numpyro/contrib/funsor/infer_util.py

```diff
@@ -109,11 +140,35 @@ def model(*args, **kwargs):
             if (scale is not None) and (not is_identically_one(scale)):
                 log_prob = scale * log_prob
 
-            log_prob = funsor.to_funsor(log_prob, output=funsor.reals(), dim_to_name=site['infer']['dim_to_name'])
-            log_factors.append(log_prob)
+            dim_to_name = site["infer"]["dim_to_name"]
+            log_prob = funsor.to_funsor(log_prob, output=funsor.reals(), dim_to_name=dim_to_name)
+
+            time_dim = None
+            for dim, name in dim_to_name.items():
+                if name.startswith("_time"):
+                    time_dim = funsor.Variable(name, funsor.domains.bint(site["value"].shape[dim]))
+                    time_to_factors[time_dim].append(log_prob)
+                    time_to_init_vars[time_dim] |= frozenset(
+                        s for s in dim_to_name.values() if s.startswith("_init"))
+                    break
+            if time_dim is None:
+                log_factors.append(log_prob)
+
             sum_vars |= frozenset({site['name']})
```

Comment:
Don't add `site['name']` to `sum_vars` here if it was observed

Reply:
Thanks, I just made the commit.